From 9919113d84310634d7acf21f8ea3d59b0d4e9d76 Mon Sep 17 00:00:00 2001
From: Ivan Mironov <mironov.ivan@gmail.com>
Date: Wed, 26 Aug 2026 13:27:48 +0200
Subject: [PATCH 1/3] Merge libcamera controls instead of trying to replace
 them

This fixes following errors:

	ERROR Camera camera.cpp:1351 Overwriting Request::controls() is not allowed
	indigo_server: indigo_ccd_rpi[StartCapture:806]: Can't queue request
---
 indigo_optional_drivers/ccd_rpi/indigo_ccd_rpi.cpp | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/indigo_optional_drivers/ccd_rpi/indigo_ccd_rpi.cpp b/indigo_optional_drivers/ccd_rpi/indigo_ccd_rpi.cpp
index 370b29877af5..1bfcb6ab1b6e 100644
--- a/indigo_optional_drivers/ccd_rpi/indigo_ccd_rpi.cpp
+++ b/indigo_optional_drivers/ccd_rpi/indigo_ccd_rpi.cpp
@@ -921,9 +921,7 @@ int RPiCamera::QueueRequest(Request *request) {
 	if (!m_cameraStarted) {
 		return -1;
 	}
-	{
-		request->controls() = std::move(m_controls);
-	}
+	request->controls().merge(m_controls, ControlList::MergePolicy::OverwriteExisting);
 	return m_camera->queueRequest(request);
 }
 
-- 
2.55.0

